src/include/duckdb/storage/data_table.hpp

```diff
@@ -77,6 +77,8 @@ class DataTable {
 	//! Constructs a DataTable as a delta on an existing data table but with one column changed type
 	DataTable(ClientContext &context, DataTable &parent, idx_t changed_idx, const LogicalType &target_type,
 	          vector<column_t> bound_columns, Expression &cast_expr);
+	//! Constructs a DataTable as a delta on an existing data table but with one column added new constraint
+	DataTable(ClientContext &context, DataTable &parent, unique_ptr<Constraint> constraint);
```

Comment:
The `DataTable` class does not hold the constraints, no? Is there a reason to create a new instance of it? Would it not be cleaner to add a `VerifyNewConstraint` method or so that verifies the constraint against the contents of the table?

Reply:
Thanks for your comments!
I didn't find constraints in `DataTable` class. The constraints used in `VerifyXxxxConstraints` functions are passed from TableCatalogEntry argument, which is not ready in this case. Not sure if there's better approach to get the new constraint, do you have any idea?
About the `VerifyNewConstraint`, actually I was thinking to extract the code to such method when I (or someone else :-)) generalize the check to more constraints in the future. Anyway, you're right, I should refactor it now. Will let you review the change shortly.

Reply:
Just added the `VerifyNewConstraint` function. Please let me know if you have any concern.